demo/date-time-picker-validation-demos.html

```diff
@@ -0,0 +1,51 @@
+<dom-module id="date-time-picker-validation-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+    <h3>Required</h3>
+    <vaadin-demo-snippet id="date-time-picker-required">
+      <template preserve-content>
+        <vaadin-date-time-picker label="Date and time" required error-message="Please choose a value">
+        </vaadin-date-time-picker>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Custom validator</h3>
+    <p>
+      Extend <code>Vaadin.DateTimePickerElement</code> to create your own custom element,
+      then override the <code>checkValidity(value)</code> method to validate the input.
+    </p>
+    <vaadin-demo-snippet id="date-time-picker-custom-validator" ignore-ie when-defined="vaadin-date-time-picker">
```

Comment:
This demo is using `vaadin-date-picker` only, not `vaadin-date-time-picker`. Update it to use `vaadin-date-time-picker` and test that it works logically.

Reply:
What do you mean?
This demo is using `DateTimePicker` and it works logically:

As `error-message` states: `Only even numbers in date and time are allowed`, so it won't accept any values containing odd number, i.e. 2020-0**3**-02T0**3**:02.

It's checking `this.value` in `checkValidity` what does mean that `value` should exist for the component (both `date-picker` and `time-picker` have values).

![dtp](https://user-images.githubusercontent.com/29372419/73925994-939e1200-48d7-11ea-943b-9b154535888b.gif)

Reply:
I don't know what happened. I was sure I saw it only using date-picker, I guess I had also date-picker repo open and was accidentally looking at the wrong repo. Sorry about that. 😄

Reply:
Ok this "Custom validator" demo seems to work fine but it's confusing. The label "Select a working range" doesn't make any sense to me (there is no "range") and also requiring all digits in the value to be "even" is very weird. This should be changed to some more realistic use case and should have a sensible label.

Reply:
Updated the demo to use more realistic use case.

Reply:
That's great demo! Thanks.